src/Microsoft.TemplateEngine.Tasks/Tasks/LocalizeTemplates.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Extensions.Logging;
+using Microsoft.TemplateEngine.Tasks.Utilities;
+using Microsoft.TemplateEngine.TemplateLocalizer.Core;
+
+namespace Microsoft.TemplateEngine.Tasks
+{
+    /// <summary>
+    /// A task that exposes template localization functionality of
+    /// Microsoft.TemplateEngine.TemplateLocalizer through MSBuild.
+    /// </summary>
+    public sealed class LocalizeTemplates : Build.Utilities.Task, ICancelableTask
+    {
+        private volatile CancellationTokenSource? _cancellationTokenSource;
+
+        /// <summary>
+        /// Gets or sets the path to the template to be localized.
+        /// </summary>
+        [Required]
+        public string? TemplateFolder { get; set; }
+
+        /// <summary>
+        /// Gets or sets the value indicating whether the subfolders
+        /// should be searched for templates.
+        /// </summary>
+        [Required]
+        public bool SearchSubfolders { get; set; } = true;
+
+        /// <summary>
+        /// Gets or sets the list of supported languages for which
+        /// localization files will be created.
+        /// </summary>
+        public string[]? Languages { get; set; }
+
+        public override bool Execute()
+        {
+            if (string.IsNullOrWhiteSpace(TemplateFolder))
+            {
+                Log.LogError(LocalizableStrings.command_localize_log_templatefolder_not_set);
+                return false;
+            }
+
+            List<string> templateJsonFiles = GetTemplateJsonFiles(TemplateFolder!, SearchSubfolders).ToList();
+
+            if (templateJsonFiles.Count == 0)
+            {
+                Log.LogError(LocalizableStrings.command_localize_log_templateJsonNotFound, TemplateFolder);
```

Comment:
The idea is, if you set this up, but the templates are somehow missing (because you moved them somewhere else, for instance), you won't notice it if there is no error. If you have this enabled, it is expected that you have at least one template in your repo. This was also an error in template localizer tool for the same reason.